test/api_image_save_and_load_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"net/url"
+
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/test/environment"
+	"github.com/alibaba/pouch/test/request"
+
+	"github.com/go-check/check"
+)
+
+// APIImageSaveLoadSuite is the test suite for image save and load API.
+type APIImageSaveLoadSuite struct{}
+
+func init() {
+	check.Suite(&APIImageSaveLoadSuite{})
+}
+
+// SetUpTest does common setup in the beginning of each test.
+func (suite *APIImageSaveLoadSuite) SetUpTest(c *check.C) {
+	SkipIfFalse(c, environment.IsLinux)
+
+	PullImage(c, helloworldImage)
+}
+
+// TestImageSaveLoadOk tests saving and loading images are OK.
+func (suite *APIImageSaveLoadSuite) TestImageSaveLoadOk(c *check.C) {
+	before, err := request.Get("/images/" + helloworldImage + "/json")
+	c.Assert(err, check.IsNil)
+	CheckRespStatus(c, before, 200)
+	gotBefore := types.ImageInfo{}
+	err = request.DecodeBody(&gotBefore, before.Body)
+	c.Assert(err, check.IsNil)
+
+	q := url.Values{}
+	q.Set("name", helloworldImage)
+	query := request.WithQuery(q)
+	resp, err := request.Get("/images/save", query)
+	c.Assert(err, check.IsNil)
+	defer resp.Body.Close()
```

Comment:
ok, will done soon.

Reply:
I mean that save the body data into `tmpFile` not the var

Reply:
@fuweid I don't think we should create a new file to store the response data stream, because this is a API test, not client test. In API test I think we handle the http response data is enough. And both `save` and `load` commands handle the data stream to tar file(or tar file to data stream) in client side, not the http response side. And in other API test, such as `api_image_inspect_test` and `api_container_inspect_test`, we just use the HTTP response data to check the correctness of this API function. WDYT?

Reply:
client test includes the api test, I think. cli wrap the client to provide better UEX. for this cases, the api test is same to the cli test. However, you connect two http connections, which is pretty hacking in testing. So I want to save the data in the local and send it again.